libs/jsonforms-components/src/lib/Controls/FormStepper/FormStepperControl.tsx

```diff
@@ -57,28 +60,70 @@ export const getFormFieldValue = (scope: string, data: object) => {
       }
       currentValue = currentValue[key];
     }
-    return typeof currentValue === 'object' ? '' : currentValue;
+    return Array.isArray(currentValue)
+      ? currentValue[currentValue.length - 1]
+      : typeof currentValue === 'object'
+      ? ''
+      : currentValue;
   } else {
     return '';
   }
 };
 
-export const renderFormFields = (elements: UISchemaElement[] | (Category | Categorization)[], data: object) =>
+export const renderFormFields = (
```

Comment:
Any time you see large functions like this inside a file, please try to extract it into its own, testable, reusable file.  Start thinking about the size of files - generally they should not be larger than a couple of hundred lines of code, at most.  Moving chunks of complex code into separate files makes things much easier to digest for people reviewing, and maintaining this code.  Do you know about separation of concerns?

This function looks quite complex, and should have a large number of unit tests to cover the various bits of logic.  Thats much easier to do if the function can be isolated.